python/main.py

```diff
@@ -23,9 +25,51 @@ def root():
     return {"message": "Hello, world!"}
 
 @app.post("/items")
-def add_item(name: str = Form(...)):
-    logger.info(f"Receive item: {name}")
-    return {"message": f"item received: {name}"}
+async def add_item(name: str = Form(...), category: str = Form(...), image: UploadFile = File(...)):
+    logger.info(f"Receive item, name: {name}, category: {category}")
+
+    # hash image
+    imageContent = await image.read()
+    hashedImage = hashlib.sha256(imageContent).hexdigest()
+    hashedImageName = hashedImage + os.path.splitext(image.filename)[1]
+
+    # update items.json file
+    newItem = {
+        'name': name,
+        'category': category,
+        'image_filename': hashedImageName
+    }
+
+    with open("items.json", 'r') as itemsFile:
+        allItems = json.load(itemsFile)
+        if not "items" in allItems:
+            raise HTTPException(status_code=502, detail="items.json file is corrupted")
```

Comment:
You might wanna use this library `http` so you don't have to type status codes manually: https://docs.python.org/3/library/http.html